### ISSUE TITLE
[Question] Atmega32u4 Connection

### ISSUE THREAD

**Issue by Tornquist** [qmk/qmk_toolbox#93]
I am working on a custom macropad powered by an atmega32u4. I’m struggling to have it recognized by the computer and was wondering if you have any suggestions for verifying the various layers. I realize this isn’t an issue, and am happy to move this if there is a better place to ask.

## Setup

* atmega32u4-au
    * Uses external clock
    * Should come flashed with the atmel bootloader
* Circuit
    * Atmega with datasheet-recommended usb schematic
    * Decoupling capacitors on all power/ground pairs
    * 16 MHz timing crystal
    * Reset on push button. Pulled high normally
    * HWB connected to gnd through 10k r
* Computer
    * macOS with up to date qmk toolbox and dependencies
    * Have used this computer to flash a teensy board and other boards in the past

## Status

* I cannot see the device from dfu-programmer
* No unknown usb devices are listed

## Questions

* Do you have any recommended things to try to verify the device exists/there is a bootloader?
* My understanding is that the out of the box atmel bootloader is fine, and that did-programmer is used through the toolbox to flash the device. Is that understanding correct?

**Comment by fauxpark** [comment]
Hi! This is indeed more a place to file issues about the QMK Toolbox itself, but since you're here... 😄

Anyway, I did some sleuthing and found your macropad repo. I don't have EAGLE but I loaded the  schematic into KiCad. It seems you have the D+ and D- lines swapped around on their way to the MCU:
![no](https://user-images.githubusercontent.com/4781841/59770279-1b082800-92eb-11e9-8d7c-92b10dc7af8a.PNG)

They should look more like this:
![ok](https://user-images.githubusercontent.com/4781841/59770274-18a5ce00-92eb-11e9-86cc-2db2a855ebd5.PNG)

To answer your questions: On macOS there are no finicky drivers to mess around with so the Toolbox should see it on pressing the reset switch.
dfu-programmer is indeed used by the Toolbox to flash ATmega32U4s, and they come from the factory with the stock Atmel USB DFU bootloader flashed already.
